### PR TITLE
Update kubeadm reference for v1.30

### PR DESCRIPTION
**Pull request by tengqm**

### CODE REVIEW

File: content/en/docs/reference/setup-tools/kubeadm/generated/kubeadm_certs_check-expiration.md [hunk 1]
```diff
@@ -30,6 +30,13 @@ kubeadm certs check-expiration [flags]
 </colgroup>
 <tbody>
 
+<tr>
+<td colspan="2">--allow-missing-template-keys&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default: true</td>
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/generated/kubeadm_certs_check-expiration.md:34]: this should not be a kubeadm flag, we need to understand what is going on here.
https://github.com/search?q=repo%3Akubernetes%2Fkubernetes%20allow-missing-template-keys&type=code

File: content/en/docs/reference/setup-tools/kubeadm/generated/kubeadm_certs_check-expiration.md [hunk 2]
```diff
@@ -44,6 +51,13 @@ kubeadm certs check-expiration [flags]
 <td></td><td style="line-height: 130%; word-wrap: break-word;"><p>Path to a kubeadm configuration file.</p></td>
 </tr>
 
+<tr>
+<td colspan="2">-o, --experimental-output string&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default: "text"</td>
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/generated/kubeadm_certs_check-expiration.md:55]: ditto

File: content/en/docs/reference/setup-tools/kubeadm/generated/kubeadm_certs_check-expiration.md [hunk 3]
```diff
@@ -58,6 +72,13 @@ kubeadm certs check-expiration [flags]
 <td></td><td style="line-height: 130%; word-wrap: break-word;"><p>The kubeconfig file to use when talking to the cluster. If the flag is not set, a set of standard locations can be searched for an existing kubeconfig file.</p></td>
 </tr>
 
+<tr>
+<td colspan="2">--show-managed-fields</td>
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/generated/kubeadm_certs_check-expiration.md:76]: ditto.

**Review by neolit123 — COMMENTED** [review]
https://github.com/kubernetes/kubernetes/pull/123372/files#diff-6799fb75a0f7b206ad9df4784047d69309b89001778ff5b3be5c155431280e5fR33

@carlory @pacoxu 
the above is not correct. it forces the kubeadm check expiration command to import some flags that we don't want.

**Comment by neolit123** [comment]
> https://github.com/kubernetes/kubernetes/pull/123372/files#diff-6799fb75a0f7b206ad9df4784047d69309b89001778ff5b3be5c155431280e5fR33
> 
> @carlory @pacoxu the above is not correct. it forces the kubeadm check expiration command to import some flags that we don't want.

looks like all kubeadm commands now have these extra flags allow-missing-template-keys ,  --show-managed-fields

also wasn't --experimental-output just --output?
i'm trying to find where this change happened.

```
$ kubeadm config images list --help

Print a list of images kubeadm will use. The configuration file is used in case any images or image repositories are customized

Usage:
  kubeadm config images list [flags]

Flags:
      --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
      --config string                 Path to a kubeadm configuration file.
  -o, --experimental-output string    Output format. One of: text|json|yaml|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file. (default "text")
      --feature-gates string          A set of key=value pairs that describe feature gates for various features. Options are:
                                      EtcdLearnerMode=true|false (BETA - default=true)
                                      PublicKeysECDSA=true|false (DEPRECATED - default=false)
                                      RootlessControlPlane=true|false (ALPHA - default=false)
                                      UpgradeAddonsBeforeControlPlane=true|false (DEPRECATED - default=false)
                                      WaitForAllControlPlaneComponents=true|false (ALPHA - default=false)
  -h, --help                          help for list
      --image-repository string       Choose a container registry to pull control plane images from (default "registry.k8s.io")
      --kubernetes-version string     Choose a specific Kubernetes version for the control plane. (default "stable-1")
      --show-managed-fields           If true, keep the managedFields when printing objects in JSON or YAML format.

Global Flags:
      --add-dir-header      If true, adds the file directory to the header of the log messages
      --kubeconfig string   The kubeconfig file to use when talking to the cluster. If the flag is not set, a set of standard locations can be searched for an existing kubeconfig file. (default "/etc/kubernetes/admin.conf")
      --rootfs string       [EXPERIMENTAL] The path to the 'real' host root filesystem.
      --skip-headers        If true, avoid header prefixes in the log messages
  -v, --v Level             number for the log level verbosity

```

**Comment by neolit123** [comment]
> > https://github.com/kubernetes/kubernetes/pull/123372/files#diff-6799fb75a0f7b206ad9df4784047d69309b89001778ff5b3be5c155431280e5fR33
> > @carlory @pacoxu the above is not correct. it forces the kubeadm check expiration command to import some flags that we don't want.
> 
> looks like all kubeadm commands with structured output now have these extra flags allow-missing-template-keys , --show-managed-fields
> 
> also wasn't --experimental-output just --output?
> 
> i'm trying to find where these changes happened.
> 

never mind, it seems this has been the case:
https://github.com/carlory/kubernetes/blob/227c2e7c2b2c05a9c8b2885460e28e4da25cf558/cmd/kubeadm/app/util/output/output.go#L102-L108

so we never graduated our --experimental-output flag to just --output. i will log a ticket for 1.31 for that.
and we explicitly add the extra template / json flags.

/lgtm

**Comment by carlory** [comment]
@neolit123  The ` --allow-missing-template-keys` flag exists in 1.29.

```
(base) (⎈|kind-kind:test)➜  kubernetes git:(55019c83b0f) git checkout v1.29.4
(base) (⎈|kind-kind:test)➜  kubernetes git:(55019c83b0f) go run cmd/kubeadm/kubeadm.go config images list -h | grep out
      --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
  -o, --experimental-output string    Output format. One of: text|json|yaml|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file. (default "text")
      --one-output               If true, only write logs to their native severity level (vs also writing to each lower severity level; no effect when -logtostderr=true)
```

when I (re)implement https://github.com/kubernetes/kubernetes/pull/123372 and 
 https://github.com/kubernetes/kubernetes/pull/123578, I follow the `config images list` code style. 

According to https://github.com/kubernetes/kubernetes/pull/123372, it accepts Go template as output.

**Comment by neolit123** [comment]
yes, i got confused.
logged this issue to graduate experimental-output to output in the next 2 releases
https://github.com/kubernetes/kubeadm/issues/3045

**Pull request merged**